### PR TITLE
jaxbind: document JAXbind internal api

### LINKED ISSUE #8: Host documentation and demonstration script on github pages or readthedocs

### LINKED ISSUE #12: [JOSS 6532] Expose API docs
While the public interface functions do seem to have good docstrings, and there are nice examples of usage, the API docs are not sufficiently accessible by JOSS standards.

Related to #8 

ref: https://github.com/openjournals/joss-reviews/issues/6532

### PR DESCRIPTION
**Pull request by roth-jakob**
Adds doctrings to JAXbind interanl apis

### CODE REVIEW

File: jaxbind/jaxbind.py [hunk 1]
```diff
@@ -286,6 +400,9 @@ def _batch(args, in_axes, *, _func: FunctionType, **kwargs):
 
 
 def _call(*args, _func: FunctionType, **kwargs):
+    """JAXbind internal: helper function evaluating the JAX primitve for the
```
Comment (Edenhofer) [inline jaxbind/jaxbind.py]: Spelling mistake in `primitve`

File: jaxbind/jaxbind.py [hunk 2]
```diff
@@ -43,6 +43,27 @@
 
 
 def _exec_abstract(*args, _func: FunctionType, **kwargs):
+    """JAXbind internal: bridging the user abstract function to JAX
```
Comment (Edenhofer) [inline jaxbind/jaxbind.py]: Here and everywhere else:
* I think `JAXbind interal:` is redundant as the name of the function already indicates that it is hidden.
* user -> user-defined ?

File: jaxbind/jaxbind.py [hunk 3]
```diff
@@ -117,12 +159,38 @@ def _lowering(ctx, *args, _func: FunctionType, _platform="cpu", **kwargs):
...
 
 
 def _jvp(args, tangents, *, _func: FunctionType, **kwargs):
+    """JAXbind internal: bridging the user derivative function to JAX
+
+
+    Parameters
+    ----------
+    args : tuple of arrays
+        Position at which the jacobian is computed.
```
Comment (Edenhofer) [inline jaxbind/jaxbind.py]: Jacobian (like Bayes) is always spelled with a capital letter AFAIK

**Review by Edenhofer — COMMENTED** [review]
PR looks good. In the following are some minor comments. Feel free to immediately merge after having incorporated them.

**Comment by roth-jakob** [comment]
TODO before merging: 1: set branch to main in pages.yml. 2: set permission in Environments /github-pages /Deployment branches to main.

**Pull request merged**